e2e/playwright/st_checkbox_test.py

```diff
@@ -59,7 +59,7 @@ def test_checkbox_values_on_click(app: Page):
     for checkbox_element in checkbox_elements.all():
         checkbox_element.click(delay=50)
     # The app run needs to finish before we can check the values.
-    wait_for_app_run(app)
+    wait_for_app_run(app, 500)
```

Comment:
Is it just specifically this test? Do we have any knowledge of why this is happening with just this test?

Reply:
This test is more probably exposed to the issue since we are triggering various reruns (via the checkbox clicks) and then we are checking if the data after the reruns is correct. But the only way currently to check that a rerun is finished is the running status indicator. But that one isn't 100% reliable since there is some possibility that the rendering actually taking more time.

Reply:
Also, this test is mainly adapted from the existing Cypress test, a better version would be to rewrite this test using `expect().to_have_text()` on every single element. This would probably wait for this text to change. I will rewrite this test in that way if it keeps being flaky.